loader/loader.c

```diff
@@ -767,10 +768,241 @@ VkResult loader_add_to_dev_ext_list(const struct loader_instance *inst, struct l
+
+                char *token = strtok(parsing_string, tokenizer);
+                while (NULL != token) {
+                    uint32_t cur_count = disable_struct->additional_filters.count;
+                    enum loader_filter_string_type cur_filter_type;
+                    const char *actual_start;
+                    size_t actual_len;
+                    determine_filter_type(token, &cur_filter_type, &actual_start, &actual_len);
+                    if (cur_filter_type == FILTER_STRING_SPECIAL) {
+                        if (!strcmp(VK_LOADER_DISABLE_ALL_LAYERS_VAR_1, token) ||
+                            !strcmp(VK_LOADER_DISABLE_ALL_LAYERS_VAR_2, token) ||
+                            !strcmp(VK_LOADER_DISABLE_ALL_LAYERS_VAR_3, token)) {
+                            disable_struct->disable_all = true;
+                        } else if (!strcmp(VK_LOADER_DISABLE_IMPLICIT_LAYERS_VAR, token)) {
+                            disable_struct->disable_all_implicit = true;
+                        } else if (!strcmp(VK_LOADER_DISABLE_EXPLICIT_LAYERS_VAR, token)) {
+                            disable_struct->disable_all_explicit = true;
+                        }
+                    } else {
+                        if (actual_len > VK_MAX_EXTENSION_NAME_SIZE) {
+                            strncpy(disable_struct->additional_filters.filters[cur_count].value, actual_start,
+                                    VK_MAX_EXTENSION_NAME_SIZE);
+                        } else {
+                            strncpy(disable_struct->additional_filters.filters[cur_count].value, actual_start, actual_len);
+                        }
+                        disable_struct->additional_filters.filters[cur_count].length = actual_len;
+                        disable_struct->additional_filters.filters[cur_count].type = cur_filter_type;
+                        disable_struct->additional_filters.count++;
+                        if (disable_struct->additional_filters.count >= MAX_ADDITIONAL_FILTERS) {
+                            break;
+                        }
+                    }
+                    token = strtok(NULL, tokenizer);
+                }
+                loader_instance_heap_free(inst, parsing_string);
+            }
+        }
+        loader_free_getenv(env_var_value, inst);
+    }
+}
+
+// Check to see if the provided layer name matches any of the filter strings.
+// This will properly check against:
+//  - substrings "*string*"
+//  - prefixes "string*"
+//  - suffixes "*string"
+//  - full string names "string"
+bool check_name_matches_filter(const struct loader_instance *inst, const char *name,
+                               const struct loader_envvar_filter *filter_struct) {
+    bool ret_value = false;
+    const size_t name_len = strlen(name);
+    char *lower_name = loader_instance_heap_calloc(inst, name_len + 1, VK_SYSTEM_ALLOCATION_SCOPE_INSTANCE);
+    if (NULL != lower_name) {
```

Comment:
Should early out here too.

Reply:
I switched this to a static size variable.  I don't want to have to worry about OOM here.